### DISCUSSION TITLE
Cannot read properties of undefined (reading 'aliasKey')

### DISCUSSION THREAD

**Discussion by jido** [11ty/eleventy#3545] [Q&A]
I am probably missing something, hope you can help me.

I added a template format as such:

```
var mistigri = require("mistigri")

module.exports = function(eleventyConfig) {
  eleventyConfig.addTemplateFormats("mi")
  mistigri.options.escapeFunction = (text) => {return String(text).replaceAll("&", "&amp;").replaceAll("<", "&lt;")}
  
  eleventyConfig.addExtension("mi", {
    compile: (inputContent) => {
      return async (data) => mistigri.prrcess(inputContent, data)
    }
  })

  return {
    dir: {
      input: "src",
      output: "public",
    }
  }
}
```

Every time I make a change to my template I receive an error like:

```
[11ty] File changed: src/_includes/base.mi
[11ty] Problem writing Eleventy templates:
[11ty] Cannot read properties of undefined (reading 'aliasKey') (via TypeError)
[11ty] 
[11ty] Original error stack trace: TypeError: Cannot read properties of undefined (reading 'aliasKey')
[11ty]     at TemplateEngineManager.isAlias
```

If I make an update in `.eleventy.js`, like adding an empty line, the error goes away.

How to prevent it?

**Comment by jido** [comment]
That stopped happening. I have no idea what changed.

**Reply by Ryuno-Ki** [reply]
In case it appears again it would help if you could link to a repository (can be a minimum working example with only the parts required to show the behaviour, if you are concerned about publishing notes that ought to be private).

**Comment by jxc876** [comment]
This was the top hit for me on google.

In my case I was learning `11ty` and tried using a pug template without realizing I had to configure it first.

```shell
npm install @11ty/eleventy-plugin-pug
```

```js
export default function (eleventyConfig) {
	eleventyConfig.addPlugin(pugPlugin);
}
```

Hope this helps someone
